### ISSUE TITLE
On macOS 11.1 (Big Sur), Pydio Cells fails to start

### ISSUE THREAD

**Issue by malik-djelil** [pydio/cells#285]
On macOS 11.1 (Big Sur), Pydio Cells fails to start.
Not tested on macOS 11.0.1 (Big Sur).
Not tested on macOS 10.15.x (Catalina).
It is OK on macOS 10.14.x (Mojave).

—

CLI to start Pydio Cells:

```
sudo --login -u cells -- sh -c 'export PATH=\"/usr/local/bin:$PATH\"; cd /Users/cells/Library/Application\ Support/Pydio/cells-bin/; ulimit -n 8192; ./cells start --log=debug'
```

(Same result without my specific path export)

—

Output (result of the CLI):

```
2020-12-20T16:05:30.051+0100	DEBUG	nats	Starting nats-server version 1.3.0
2020-12-20T16:05:30.052+0100	DEBUG	nats	Go build version go1.13.15
2020-12-20T16:05:30.052+0100	DEBUG	nats	Git commit [not set]
2020-12-20T16:05:30.052+0100	DEBUG	nats	Listening for client connections on 0.0.0.0:4222
2020-12-20T16:05:30.052+0100	DEBUG	nats	Server id is q81LHaoh2gl3X8jgkAOz25
2020-12-20T16:05:30.052+0100	DEBUG	nats	Server is ready
2020-12-20T16:05:31.053+0100	DEBUG	nats	127.0.0.1:52166 - cid:1 - Client connection created
2020-12-20T16:05:32.902+0100	DEBUG	nats	127.0.0.1:52166 - cid:1 - Client connection closed
2020-12-20T16:05:34.258+0100	DEBUG	nats	127.0.0.1:52200 - cid:2 - Client connection created
2020-12-20T16:05:34.259+0100	FATAL	Could not retrieve initial services list
github.com/pydio/cells/common/log.Fatal
	/ext-go/1/src/github.com/pydio/cells/common/log/log.go:277
github.com/pydio/cells/common/registry.(*pydioregistry).maintainRunningServicesList
	/ext-go/1/src/github.com/pydio/cells/common/registry/running.go:93
github.com/pydio/cells/common/registry.(*pydioregistry).Init
	/ext-go/1/src/github.com/pydio/cells/common/registry/registry.go:135
github.com/pydio/cells/common/registry.Init
	/ext-go/1/src/github.com/pydio/cells/common/registry/registry.go:88
github.com/pydio/cells/cmd.initServices
	/ext-go/1/src/github.com/pydio/cells/cmd/root.go:235
github.com/pydio/cells/cmd.glob..func44
	/ext-go/1/src/github.com/pydio/cells/cmd/start.go:164
github.com/pydio/cells/vendor/github.com/spf13/cobra.(*Command).execute
	/ext-go/1/src/github.com/pydio/cells/vendor/github.com/spf13/cobra/command.go:745
github.com/pydio/cells/vendor/github.com/spf13/cobra.(*Command).ExecuteC
	/ext-go/1/src/github.com/pydio/cells/vendor/github.com/spf13/cobra/command.go:846
github.com/pydio/cells/vendor/github.com/spf13/cobra.(*Command).Execute
	/ext-go/1/src/github.com/pydio/cells/vendor/github.com/spf13/cobra/command.go:794
github.com/pydio/cells/cmd.Execute
	/ext-go/1/src/github.com/pydio/cells/cmd/root.go:201
main.main
	/ext-go/1/src/github.com/pydio/cells/main.go:119
runtime.main
	/usr/local/go/src/runtime/proc.go:203
2020-12-20T16:05:34.259+0100	DEBUG	nats	127.0.0.1:52200 - cid:2 - Client connection closed
```

**Comment by jthabet** [comment]
Hello @malik-djelil ,

I just tested on my mac which is running the same version as you `11.1`and did not encounter your issue.

First, could you tell me which binary did you run (cells version),
then would you mind trying with the latest dev build and reporting if it still happens.

https://download.pydio.com/pub/cells/dev/darwin-amd64/

Regards

**Comment by malik-djelil** [comment]
Hello @jthabet,

Thank you for your help!

Yes, I tested with the latest release version (updated from the CLI, since I can't access the web GUI).
```
Pydio Cells Home Edition
 Version: 	2.1.11
 Built: 	15 Dec 20 15:55 +0000
 Git commit: 	4ad2d925cecd3a0d8133d6f2346c6f7155e7bef7
 OS/Arch: 	darwin/amd64
 Go version: 	go1.13.15
```

---

With the last dev build:
```
Pydio Cells Home Edition
 Version: 	2.2.0-dev.202101150953
 Built: 	15 Jan 21 09:54 +0000
 Git commit: 	0df246c327f78e3584d28faec675e95e7c431dfb
 OS/Arch: 	darwin/amd64
 Go version: 	go1.15.5
```

...I get a different log:
( For more clarity in this post, I replaced the domain names and ports by `pydiocells.localhostname:port` and `collaboraonline.localhostname:port`. )
```
[Configs] Upgrading: renaming key services/pydio.rest.gateway.dav to services/pydio.gateway.dav
[Configs] Upgrading: renaming key services/pydio.rest.gateway.wopi to services/pydio.gateway.wopi
[Configs] Upgrading: renaming key ports/micro.api to ports/pydio.gateway.rest
[Configs] Upgrading: renaming key services/micro.api to services/pydio.gateway.rest
[Configs] Upgrading: renaming key services/pydio.api.front-plugins to services/pydio.web.statics
[Configs] Upgrading: renaming key services/pydio.api.websocket to services/pydio.gateway.websocket
[Configs] Upgrading: renaming key services/pydio.grpc.gateway.data to services/pydio.gateway.data
[Configs] Upgrading: renaming key services/pydio.grpc.gateway.proxy to services/pydio.gateway.proxy
[Configs] Upgrading: renaming key services/pydio.grpc.auth/dex/connectors to services/pydio.web.oauth/connectors
[Configs] Upgrading source keys
2021-01-15T17:07:58.778+0100	INFO	
2021-01-15T17:07:59.055+0100	INFO	✅ Created a new local CA at "/Users/cells/Library/Application Support/Pydio/cells/certs/rootCA.pem" 💥
2021-01-15T17:07:59.055+0100	INFO	✅ Created a new certificate valid for the following names 📜 - "pydio.gateway.grpc"
2021-01-15T17:07:59.055+0100	INFO	✅ The certificate is at "/Users/cells/Library/Application Support/Pydio/cells/certs/1763c734e5c385445c57f58c3a8ee7a5.pem" 
2021-01-15T17:07:59.055+0100	INFO	 and the key at "/Users/cells/Library/Application Support/Pydio/cells/certs/1763c734e5c385445c57f58c3a8ee7a5-key.pem"
2021-01-15T17:07:59.055+0100	INFO	
2021-01-15T17:07:59.055+0100	INFO	👉 If you are behind a reverse proxy, you can either install the RootCA on the proxy machine trust store, or configure your proxy to `insecure_skip_verify` for pointing to Cells.
2021-01-15T17:07:59.055+0100	INFO	👉 If you are developing locally, you may install the RootCA in your system trust store to see a green light in your browser!
2021-01-15T17:07:59.055+0100	INFO	🗒  To easily install the RootCA in your trust store, use https://github.com/FiloSottile/mkcert. Set the $CAROOT environment variable to the rootCA folder then use 'mkcert -install'
2021-01-15T17:07:59.055+0100	INFO	
2021-01-15T17:07:59.059+0100	ERROR	pydio.gateway.proxy	Could not finalize start 	{"error": "nats: invalid connection"}
2021-01-15T17:07:59.075+0100	ERROR	pydio.grpc.data.sync	Could not run 	{"error": "write tcp 127.0.0.1:63979->127.0.0.1:4222: write: protocol wrong type for socket"}
2021-01-15T17:07:59.078+0100	ERROR	pydio.gateway.data	Could not run 	{"error": "write tcp 127.0.0.1:63981->127.0.0.1:4222: write: protocol wrong type for socket"}
2021-01-15T17:07:59.078+0100	ERROR	pydio.grpc.update	Could not run 	{"error": "write tcp 127.0.0.1:63982->127.0.0.1:4222: write: protocol wrong type for socket"}
2021-01-15T17:07:59.082+0100	ERROR	pydio.grpc.statics	Could not run 	{"error": "write tcp 127.0.0.1:63984->127.0.0.1:4222: write: protocol wrong type for socket"}
2021-01-15T17:07:59.084+0100	ERROR	pydio.grpc.data.objects	Could not run 	{"error": "write tcp 127.0.0.1:63986->127.0.0.1:4222: write: protocol wrong type for socket"}
2021-01-15T17:07:59.085+0100	ERROR	pydio.grpc.docstore	Could not run 	{"error": "write tcp 127.0.0.1:63989->127.0.0.1:4222: write: protocol wrong type for socket"}
2021-01-15T17:07:59.086+0100	ERROR	pydio.grpc.log	Could not run 	{"error": "write tcp 127.0.0.1:63991->127.0.0.1:4222: write: protocol wrong type for socket"}
2021-01-15T17:07:59.089+0100	ERROR	pydio.gateway.rest	Could not run 	{"error": "write tcp 127.0.0.1:63993->127.0.0.1:4222: write: protocol wrong type for socket"}
2021-01-15T17:07:59.153+0100	INFO	Activating privacy features... done.
2021-01-15T17:07:59.154+0100	INFO	https://pydiocells.localhostname:port
2021-01-15T17:07:59.155+0100	ERROR	pydio.gateway.proxy	Could not run 	{"error": "write tcp 127.0.0.1:64003->127.0.0.1:4222: write: protocol wrong type for socket"}
2021-01-15T17:07:59.287+0100	ERROR	pydio.grpc.config	Could not run 	{"error": "write tcp 127.0.0.1:64013->127.0.0.1:4222: write: protocol wrong type for socket"}
2021-01-15T17:07:59.314+0100	ERROR	pydio.grpc.chat	Could not run 	{"error": "write tcp 127.0.0.1:64025->127.0.0.1:4222: write: protocol wrong type for socket"}
2021-01-15T17:08:01.873+0100	ERROR	pydio.grpc.meta	Could not run 	{"error": "nats: no servers available for connection"}
2021-01-15T17:08:01.874+0100	ERROR	pydio.grpc.acl	Could not run 	{"error": "nats: no servers available for connection"}
2021-01-15T17:08:01.874+0100	ERROR	pydio.grpc.user-key	Could not run 	{"error": "nats: no servers available for connection"}
2021-01-15T17:08:01.874+0100	ERROR	pydio.grpc.policy	Could not run 	{"error": "nats: no servers available for connection"}
2021-01-15T17:08:01.875+0100	ERROR	pydio.grpc.user-meta	Could not run 	{"error": "nats: no servers available for connection"}
2021-01-15T17:08:01.875+0100	ERROR	pydio.grpc.token	Could not run 	{"error": "nats: no servers available for connection"}
2021-01-15T17:08:01.944+0100	ERROR	pydio.grpc.oauth	Could not run 	{"error": "nats: no servers available for connection"}
2021-01-15 17:08:01.946377 I | write tcp 127.0.0.1:64065->127.0.0.1:4222: write: protocol wrong type for socket
2021-01-15 17:08:01.948175 I | write tcp 127.0.0.1:64087->127.0.0.1:4222: write: protocol wrong type for socket
2021-01-15 17:08:01.949695 I | write tcp 127.0.0.1:64102->127.0.0.1:4222: write: protocol wrong type for socket
2021-01-15 17:08:01.953897 I | write tcp 127.0.0.1:64121->127.0.0.1:4222: write: protocol wrong type for socket
2021-01-15T17:08:02.021+0100	ERROR	pydio.grpc.jobs	Could not run 	{"error": "write tcp 127.0.0.1:64147->127.0.0.1:4222: write: protocol wrong type for socket"}
2021-01-15T17:08:04.161+0100	INFO	pydio.gateway.proxy	Restarting proxy	{"caddyfile": "\n\n\n\n\n\n\npydiocells.localhostname:port {\n\t\n\n\t\n\tredir 303 { \n\t\tif {path} not /starting.html\n\t\t/starting.html\n\t}\n\t\n\n\tproxy /a  PENDING {\n\t\twithout /a\n\t\theader_upstream Host pydiocells.localhostname:port\n\t\theader_upstream X-Real-IP {remote}\n\t\theader_upstream X-Forwarded-Proto {scheme}\n\t}\n\tproxy /oidc PENDING {\n\t\tinsecure_skip_verify\n\t\theader_upstream Host pydiocells.localhostname:port\n\t\theader_upstream X-Real-IP {remote}\n\t\theader_upstream X-Forwarded-Proto {scheme}\n\t}\n\tproxy /io   PENDING {\n\t\theader_upstream Host pydiocells.localhostname:port\n\t\theader_upstream X-Real-IP {remote}\n\t\theader_upstream X-Forwarded-Proto {scheme}\n\t\theader_downstream Content-Security-Policy \"script-src 'none'\"\n\t\theader_downstream X-Content-Security-Policy \"sandbox\"\n\t}\n\tproxy /data PENDING {\n\t\theader_upstream Host pydiocells.localhostname:port\n\t\theader_upstream X-Real-IP {remote}\n\t\theader_upstream X-Forwarded-Proto {scheme}\n\t\theader_downstream Content-Security-Policy \"script-src 'none'\"\n\t\theader_downstream X-Content-Security-Policy \"sandbox\"\n\t}\n\tproxy /ws   PENDING {\n\t\twebsocket\n\t\twithout /ws\n\t\theader_upstream Host pydiocells.localhostname:port\n\t\theader_upstream X-Real-IP {remote}\n\t\theader_upstream X-Forwarded-Proto {scheme}\n\t}\n\tproxy /dav PENDING {\n\t\theader_upstream Host pydiocells.localhostname:port\n\t\theader_upstream X-Real-IP {remote}\n\t\theader_upstream X-Forwarded-Proto {scheme}\n\t\theader_downstream Content-Security-Policy \"script-src 'none'\"\n\t\theader_downstream X-Content-Security-Policy \"sandbox\"\n\t}\n\t\n\n\n\tproxy /grpc https://PENDING {\n\t\twithout /grpc\n\t\tinsecure_skip_verify\n\t}\n\t\n\trewrite {\n\t\tif {>Content-type} has \"application/grpc\"\n\t\tto /grpc/{path}\n\t}\n\n\n\tredir 302 {\n\t\tif {>Content-type} not_has \"application/grpc\"\n\t\tif {path} is /\n\t\t/ /login\n\t}\n\t\n\t\n\t\n        proxy /wopi/ PENDING {\n            transparent\n\t\t\theader_upstream Host pydiocells.localhostname:port\n\t\t\theader_upstream X-Real-IP {remote}\n\t\t\theader_upstream X-Forwarded-Proto {scheme}\n        }\n\n        proxy /loleaflet/ https://collaboraonline.localhostname:port/loleaflet {\n            transparent\n            insecure_skip_verify\n            without /loleaflet/\n        }\n\n        proxy /hosting/discovery https://collaboraonline.localhostname:port/hosting/discovery {\n            transparent\n            insecure_skip_verify\n            without /hosting/discovery\n        }\n\n        proxy /lool/ https://collaboraonline.localhostname:port/lool/ {\n            transparent\n            insecure_skip_verify\n            websocket\n            without /lool/\n        }\n    \n\t\n\t\n\trewrite {\n\t\tif {path} not_starts_with \"/a/\"\n\t\tif {path} not_starts_with \"/oidc/\"\n\t\tif {path} not_starts_with \"/io\"\n\t\tif {path} not_starts_with \"/data\"\n\t\tif {path} not_starts_with \"/ws/\"\n\t\tif {path} not_starts_with \"/plug/\"\n\t\tif {path} not_starts_with \"/dav\"\n\t\t\n\t\tif {path} not_starts_with \"/wopi/\"\n\t\t\n\t\tif {path} not_starts_with \"/loleaflet/\"\n\t\t\n\t\tif {path} not_starts_with \"/hosting/discovery\"\n\t\t\n\t\tif {path} not_starts_with \"/lool/\"\n\t\t\n\t\tif {path} not_starts_with \"/public/\"\n\t\tif {path} not_starts_with \"/user/reset-password\"\n\t\tif {path} not_starts_with \"/robots.txt\"\n\t\tto {path} {path}/ /login\n\t}\n\n\troot \"/Users/cells/Library/Application Support/Pydio/cells/static/maintenance\"\n\n\t\n\ttls \"/usr/local/etc/httpd/server.crt\" \"/usr/local/etc/httpd/server.key\"\n\terrors \"/Users/cells/Library/Application Support/Pydio/cells/logs/caddy_errors.log\"\n}\n\n\n\n\n\t"}
2021-01-15T17:08:04.664+0100	INFO	pydio.gateway.proxy	Restart done
```
The block [ `Restarting proxy` - `Restart done` ] is repeated endlessly...

**Comment by malik-djelil** [comment]
I just tested with today's new dev build (January 18).
```
Pydio Cells Home Edition
 Version: 	2.2.0-dev.202101181652
 Built: 	18 Jan 21 16:53 +0000
 Git commit: 	9ba4d0d47e2e91ab4443ce5cd4dda60493c9fd3d
 OS/Arch: 	darwin/amd64
 Go version: 	go1.15.5
```

There are still the same errors in the log.

**Comment by jthabet** [comment]
on your first error message, we could see a crash, now we can observe that the error is about a wrong protocol type for socket, it could point to a binding issue,

- does the user that you use to run Cells have specific permissions ?
- do you make sure to clean your previous install (removing the folder pydio under `/Library/Application Support/Pydio/`) before re installing?

Could you try a simple install with the defaults settings to see if it happens, try to not put many arguments in the command, if possible a simple `./cells configure` will start the installation wizard and guide you.

**Comment by malik-djelil** [comment]
For information, I have also tested these versions:

```
Pydio Cells Home Edition
 Version:   2.2.0
 Built:     28 Jan 21 08:50 +0000
 Git commit:    5435dca7487e314483e2c6e2e3cf9ae7f6bf24be
 OS/Arch:   darwin/amd64
 Go version:    go1.15.5
```

```
Pydio Cells Home Edition
 Version:   2.2.1-dev.202101291514
 Built:     29 Jan 21 15:15 +0000
 Git commit:    85b1992c2f1af307ec09a3afccc27452b6404b62
 OS/Arch:   darwin/amd64
 Go version:    go1.15.5
```

The log is different, but the issue is still here.

---

> does the user that you use to run Cells have specific permissions?

It is a standard user of macOS. He has the permissions to run Pydio Cells (like on Mojave where it worked).

> do you make sure to clean your previous install (removing the folder pydio under /Library/Application Support/Pydio/) before re installing?

No.
For a release version, I update with the CLI like this: `./cells update --version=2.2.0`.
For a dev build, I download the cells binary (as you have indicated to me) and I replace the old one on my install.

---

I need to investigate further...
Otherwise, I will try with a clean installation, but it is not a solution for me. I would like to preserve my Pydio environment (files, users, DB...).

**Comment by malik-djelil** [comment]
ISSUE RESOLVED !

1. The problem quickly mentioned in the previous comment and related to the access to the files is caused by a bad configuration of the hostname of my server (it was done for security tests with another application). So, the files are displayed correctly (with a correct server configuration).

2. I have however a problem of upload (import, creation, modification, deletion of file or directory). To avoid this thread to be confused, I will create another ticket with the logs if I don't find the solution or if I think the problem comes from Pydio Cells. 

3. In CONCLUSION, **the Pydio Cells startup issue (with errors in the logs) has been fixed with the 2.2.0 (release) to 2.2.2 (release) update.** For this specific problem (related to this ticket), I think this thread can be closed (issue resolved).

**Comment by bsinou** [comment]
Hey @malik-djelil, thanks for reporting. I'll close the ticket.

Yet, may I please ask you to first describe your other problem in the Forum. And only open an issue here if / when we are sure it is a bug ? 
It would be nice, thanks.
